instance/import.go

```diff
@@ -54,104 +59,123 @@ func (s *Store) UpdateImportTask(w http.ResponseWriter, r *http.Request) {
 	ctx := r.Context()
 	vars := mux.Vars(r)
 	instanceID := vars["instance_id"]
+	eTag := getIfMatch(r)
 	logData := log.Data{"instance_id": instanceID}
 	defer r.Body.Close()
 
-	updateErr := func() *taskError {
-		tasks, err := unmarshalImportTasks(r.Body)
-		if err != nil {
-			log.Event(ctx, "failed to unmarshal request body to UpdateImportTasks model", log.ERROR, log.Error(err), logData)
-			return &taskError{err, http.StatusBadRequest}
+	handleError := func(updateErr *taskError) {
+		log.Event(ctx, "updateImportTask endpoint: request unsuccessful", log.ERROR, log.Error(updateErr), logData)
+		http.Error(w, updateErr.Error(), updateErr.status)
+	}
+
+	tasks, err := unmarshalImportTasks(r.Body)
+	if err != nil {
+		log.Event(ctx, "failed to unmarshal request body to UpdateImportTasks model", log.ERROR, log.Error(err), logData)
+		handleError(&taskError{err, http.StatusBadRequest})
+		return
+	}
+
+	instance, err := s.GetInstance(instanceID, eTag)
```

Comment:
Does this need wrapping with :
AcquireInstanceLock()
etc as done in dimension/dimension.go ... about line 45'ish ?

Reply:
if the ETag changes between any of these calls:
- GetInstnace and UpdateImportObservationsTaskState
- UpdateImportObservationsTaskState and UpdateBuildHierarchyTaskState
- UpdateBuildHierarchyTaskState and UpdateBuildSearchTaskState
Then it will fail. However, in this case, I think it makes sense to add the lock, to be confident that if we start changing something, no other call will change anything else meanwhile.